### DISCUSSION TITLE
Retrieve params after optimization

### DISCUSSION THREAD

**Discussion by carolinafernandezp** [jonescompneurolab/hnn-core#578] [Q&A] [answered]
Hi,

After optimization with hnn_core, how can I access the new, optimized, parameters? 

Thanks

**Comment by rythorpe** [comment, marked as answer]
The best way would be to access the `net.external_drives` attribute, which contains the dynamic properties of each drive via e.g. `net.external_drives['evprox1']['dynamics']`. For synaptic weight parameters, you can inspect the `net.connectivity` attribute, which is itself a list of all the connections within and onto the network. Since `net.connectivity` is itself a list of all the connections defined in the network, you can use [`pick_connection`](https://jonescompneurolab.github.io/hnn-core/dev/generated/hnn_core.pick_connection.html#hnn_core.pick_connection) to find the index associated with the drive of interest.

Now that you bring this up, we should probably add this to the optimization example....

**Comment by jasmainak** [comment]
maybe @carolinafernandezp easy pull request for you if you want to try? It can count towards the 1 pull request requirement if you end up applying to GSoC

**Reply by carolinafernandezp** [reply]
Ok I can definitely do that!

**Reply by jasmainak** [reply]
let us know if you need help. Here is the contributing guide: https://jonescompneurolab.github.io/hnn-core/dev/contributing.html

I'm sure Andy can also help if you're stuck :)

**Comment by jasmainak** [comment]
@carolinafernandezp are you still planning to apply for GSoC? We just submitted our application as an organization. After that, it will be time for candidates to start applying. This is the right moment to make a pull request if you are still planning on applying!

**Comment by carolinafernandezp** [comment]
Hi Mainak,

Thanks so much for your reminders and sorry for getting back to you so late! Things have been busy at the lab but I'm finally getting to making the pull request. I'm leaning towards the IO routines project (I just emailed Dr. Jones about it).

Thanks again 🙂


Carolina

________________________________
From: Mainak Jas ***@***.***>
Sent: Tuesday, January 17, 2023 12:01 PM
To: jonescompneurolab/hnn-core ***@***.***>
Cc: Fernandez, Carolina ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [jonescompneurolab/hnn-core] Retrieve params after optimization (Discussion #578)

CAUTION: This email originated from outside the organization. DO NOT CLICK ON LINKS or OPEN ATTACHMENTS unless you know and trust the sender.

@carolinafernandezp<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcarolinafernandezp&data=05%7C01%7Ccxf418%40miami.edu%7C0831a201b8164f6cf79b08daf8c593fa%7C2a144b72f23942d48c0e6f0f17c48e33%7C0%7C0%7C638095824787460412%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=muuDekuO6i5EVLip%2F%2BKpVrX0whi%2FWaWu9TzbXOzjzgI%3D&reserved=0> are you still planning to apply for GSoC? We just submitted our application as an organization. After that, it will be time for candidates to start applying. This is the right moment to make a pull request if you are still planning on applying!
